### PR TITLE
GRAPE-19395

### PR DESCRIPTION
**Pull request by AleshaOleg**

### CODE REVIEW

**Review by pseudonymcp — CHANGES_REQUESTED** [review]
What's missing is that if a search field is invalid, then the help text has an id the search field can reference to via aria-describedby, and aria-describedby of the search field connects to the error message.

Btw: In Storybook I'm getting an error on the canvas when I try to enter a validationHelp string in the Base story's Controls.


Update:  
- Please check and correct also the font size inside the input.

File: web/src/components/text-input/generic.tsx [hunk 1]
```diff
@@ -16,6 +17,13 @@ export interface TextInputProps extends AriaTextFieldOptions {
   description?: string
   validationHelp?: string
   isRequired?: boolean
+  maxLength?: number
+<<<<<<< HEAD
+=======
+  minLength?: number
+>>>>>>> 1d4b1a305bfa5a84a6d1a09039e149bc34d6a4ae
```
Comment (sk7) [inline web/src/components/text-input/generic.tsx:24]: broken :D

**Review by pseudonymcp — CHANGES_REQUESTED** [review]
Found some issues:
- font-size inside input is wrong (computed 13.3333px)
- the `necessityIndicator` prop isn't documented: how to define if "required" and "optional" suffix should be shown or hidden? → "A required or optional field can either be denoted with text added to the end of the label or have no indication at all."
- `maxLength` prop should be removed (if it's set it would overlap with the remove-icon)
- `type` prop isn't needed as Search Field has only type=search (is ignored anyway, so please remove it)
- `min` and `max` props should be removed for Search Field as they have no effect and aren't needed

**Review by pseudonymcp — APPROVED** [review]
Looks good now for me 👍

**Pull request merged**